src/fragments/lib/graphqlapi/js/optimistic-ui.mdx

```diff
@@ -0,0 +1,1049 @@
+  return { previousRealEstateProperty, newRealEstateProperty };
+},
+// If the mutation fails, use the context we returned above
+onError: (err, newRealEstateProperty, context) => {
+  console.error("Error deleting record:", err, newRealEstateProperty);
+  if (context?.previousRealEstateProperty) {
+    queryClient.setQueryData<RealEstateProperty>(
+      ["realEstateProperties", context.newRealEstateProperty.id],
+      context.previousRealEstateProperty
+    );
+  }
+},
+// Always refetch after error or success:
+onSettled: (newRealEstateProperty) => {
+  queryClient.invalidateQueries({
+    // TODO: fix type
+    // @ts-ignore
+    queryKey: ["realEstateProperties", newRealEstateProperty.id],
+  });
+},
+});
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+// TODO - add JS example after approval of TS code snippet
+```
+
+</Block>
+
+</BlockSwitcher>
+
+## Loading and error states for optimistically rendered data
+
+Both `useQuery` and `useMutation` return `isLoading` and `isError` states that indicate the current state of the query or mutation. You can use these states to render loading and error indicators.
+
+In addition to operation-specific loading states, TanStack Query provides a [`useIsFetching` hook](https://www.tanstack.com/query/v4/docs/react/guides/background-fetching-indicators#displaying-global-background-fetching-loading-state). For the purposes of this demo, we show a global loading indicator in the [Complete Example](#complete-example) when *any* queries are fetching (including in the background) in order to help visualize what TanStack is doing in the background:
+
+```js
+function GlobalLoadingIndicator() {
+  const isFetching = useIsFetching();
+  return isFetching ? <div style={styles.globalLoadingIndicator}></div> : null;
+}
+```
+
+For more details on advanced usage of TanStack Query hooks, see the [TanStack documentation](https://tanstack.com/query/latest/docs/react/guides/mutations).
+
+The following example demonstrates how to use the state returned by TanStack to render a loading indicator while a mutation is in progress, and an error message if the mutation fails. For additional examples, see the [Complete Example](#complete-example) below.
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+<>
+  {updateMutation.isError &&
+  updateMutation.error instanceof Error ? (
+    <div>An error occurred: {updateMutation.error.message}</div>
+  ) : null}
+
+  {updateMutation.isSuccess ? (
+    <div>Real Estate Property updated!</div>
+  ) : null}
+
+  <button
+    onClick={() =>
+      updateMutation.mutate({
+        id: realEstateProperty.id,
+        address: `${Math.floor(
+          1000 + Math.random() * 9000
+        )} Main St`,
+      })
+    }
+  >
+    Update Address
+  </button>
+</>
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+// TODO - add JS example after approval of TS code snippet
+```
+
+</Block>
+
+</BlockSwitcher>
+
+## Complete examples
```

Comment:
@david-mcafee Should this just be "Complete example" (singular)?

Reply:
I made this plural due to the fact that there is both a TS and JS example, but singular makes more sense here!